### ISSUE TITLE
Imported ec_deployment resource displays incorrect version

### ISSUE THREAD

**Issue by f1nzer** [elastic/terraform-provider-ec#614]
## Readiness Checklist

- [x] I am running the latest version
- [x] I checked the documentation and found no answer
- [x] I checked to make sure that this issue has not already been filed
- [x] I am reporting the issue to the correct repository (for multi-repository projects)

## Expected Behavior
`terraform state import` imports a correct version of ec_deployment

## Current Behavior

The version it imports is incorrect: it displays 8.4.3, instead of 8.7.0
![image](https://user-images.githubusercontent.com/1970236/229204552-569f43db-00b9-4e9f-998f-1e543b91cb34.png)

## Context

I had an ec_deployment of version 8.4.3.
I've updated my cluster manually to version 8.7.0 using elastic cloud dashboard.
Now I'm trying to import terraform state using latest ec provider (0.6.0).

I've tried to update it to 8.7.0 by changing my tf configuration (my expectation was that it will be a no-op):
```tf
resource "ec_deployment" "main" {
...
  version                = "8.7.0"
```

terraform plan partial output:
```
~ version                = "8.4.3" -> "8.7.0"
```

Unfortunately, `terraform apply` command throws exception:
```
│ Error: Provider produced inconsistent result after apply
│
│ When applying changes to ec_deployment.main, provider
│ "provider[\"registry.terraform.io/elastic/ec\"]" produced an unexpected new
│ value: .version: was cty.StringVal("8.7.0"), but now
│ cty.StringVal("8.4.3").
│
│ This is a bug in the provider, which should be reported in the provider's
│ own issue tracker.
╵
```

## Your Environment

* Version used: 0.6.0
* Running against Elastic Cloud
* Terraform: 1.4.2
* Operating System and version: Windows 10

**Comment by f1nzer** [comment]
Ok, I figured out what was the problem.

According to ElasticCloud API (/deployments/deploymentId) I had Enterprise Search of version 8.4.3, while all other resources are 8.7.0.

Many months ago, I had an Enterprise Search that was disabled (removed).
For some reason, it was not updated with other resources on the recent update to 8.7.0 - it is ok, because it was disabled/removed.

There is a part of response from /deployments endpoint:
```json
"enterprise_search": [
  "status": "stopped",
  "plan_info": {
    "current": {
        "plan": {
            "enterprise_search": {
                "version": "8.4.3"
            }
        }
    }
}
```

But, it looks like tf ec provider ignores the fact that it was removed and probably uses the minimal version from all resources. I haven't checked the code yet.

My fix was to re-enable Enterprise Search (it automatically triggers an update to 8.7.0) and after that ec provider works as expected. 
![image](https://user-images.githubusercontent.com/1970236/229210258-c5a3ce94-5ebf-46f5-ba6a-c303ac0ac924.png)
